src/json.ts

```diff
@@ -34,6 +41,119 @@ export type Json =
   | Json[]
   | { [prop: string]: Json };
 
+/**
+ * A helper type to make properties with `undefined` in their type optional, but
+ * not `undefined` itself.
+ *
+ * @example
+ * ```ts
+ * type Foo = ObjectOptional<{ foo: string | undefined }>;
+ * // Foo is equivalent to { foo?: string }
+ * ```
+ */
+export type ObjectOptional<Schema extends Record<string, unknown>> = {
```

Comment:
It seems like this type essentially converts properties that _should_ be optional (based on the fact that they can be `undefined`) so that they actually are. I don't know, however, that I would be able to guess that from this name. I see that Superstruct already has a type called `Optionalize`. What does that type do that this doesn't do? What do you think about a name like `NormalizeOptionals`? Then again, if this name is patterned after something else I'm not aware of, let me know.

Reply:
Superstruct's `Optionalize` doesn't work with exact optionals. This is basically a copy of that, but also checks if the property has a `ExactOptionalGuard`, in which case the type will be inferred as `key?: Type`, instead of `key?: Type | undefined`. It's a bit hard to read 😅 I don't expect consumers of `utils` to use this type directly, so I don't think the name really matters?

Reply:
Er, I guess that's a silly question. I can see how they would be useful, so never mind!

Reply:
I think it's good practice to export any "helper" types used. There are rare cases (usually in tests) where you may need to reproduce a certain type, and it helps if those are exported.